### DISCUSSION TITLE
is possibel to use pack command to create a offline rock

### DISCUSSION THREAD

**Discussion by miniboxhaha** [keplerproject/luarocks#1724] [Q&A] [answered]
in my case, i depended luajwtjitsi, and luajwtjitsi depended other modules.
when i use pack command, the files in pack only contains handler.lua,schema.lua
what if i need to deploy the rock file on a host without internet?
```
package = "role_based_with_jwt"
version = "dev-1"
source = {
   url = "*** please add URL for source tarball, zip or repository here ***"
}
description = {
   homepage = "*** please enter a project homepage ***",
   license = "*** please specify a license ***"
}
dependencies = {
   "luajwtjitsi >= 3.0"
}
build = {
   type = "builtin",
   modules = {
      handler = "handler.lua",
      schema = "schema.lua"
   }
}
```

**Comment by Tieske** [comment, marked as answer]
luarocks cannot pack a dependency tree afaik. So you'd have to pack all dependencies individually.
Installing can be done individually as well, or by createing a temporary luarocks server (with all rocks), and only installing the target rock (where luarocks will pull them from your temp server)

**Reply by miniboxhaha** [reply]
understood，thanks
